app/(auth)/index.tsx

```diff
@@ -0,0 +1,166 @@
+import { Link, router, useFocusEffect } from 'expo-router';
+import { useCallback, useState } from 'react';
+import {
+  Alert,
+  Pressable,
+  SafeAreaView,
+  StyleSheet,
+  Text,
+  TextInput,
+  View,
+} from 'react-native';
+import { colors } from '../../constants/colors';
+import type { LoginResponseBodyPost } from '../api/login+api';
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    backgroundColor: colors.background,
+    alignItems: 'center',
+    width: '100%',
+  },
+  loginInputContainer: {
+    backgroundColor: colors.cardBackground,
+    borderRadius: 12,
+    padding: 12,
+    marginBottom: 16,
+    width: '100%',
+  },
+  label: {
+    fontSize: 18,
+    fontFamily: 'Poppins_400Regular',
+    color: colors.text,
+    marginBottom: 8,
+  },
+  input: {
+    color: colors.text,
+    backgroundColor: colors.background,
+    borderColor: colors.textSecondary,
+    borderWidth: 1,
+    borderRadius: 8,
+    padding: 8,
+    marginBottom: 16,
+    fontSize: 16,
+    fontFamily: 'Poppins_400Regular',
+  },
+  inputFocused: {
+    borderColor: colors.white,
+  },
+  promptTextContainer: {
+    flexDirection: 'row',
+    justifyContent: 'center',
+    alignItems: 'center',
+    gap: 4,
+  },
+  button: {
+    marginTop: 30,
+    backgroundColor: colors.text,
+    padding: 12,
+    borderRadius: 12,
+    shadowColor: colors.white,
+    shadowOffset: { width: 0, height: 2 },
+    shadowOpacity: 0.2,
+    shadowRadius: 4,
+    elevation: 4,
+    width: '100%',
+  },
+  text: {
+    fontFamily: 'Poppins_400Regular',
+    color: colors.cardBackground,
+    textAlign: 'center',
+    fontSize: 18,
+  },
+});
+
+export default function Login() {
```

Comment:
Should the filename match the component name? (but lowercase)

Reply:
The idea here is that users should land on the login page when they visit the application, and Expo would recognize the `index.tsx` page to be that page

Reply:
<img width="1920" alt="Screenshot 2024-11-03 at 11 54 00" src="https://github.com/user-attachments/assets/0218800a-d956-4037-9d53-63e497f95f2d">
<img width="455" alt="Screenshot 2024-11-03 at 11 54 43" src="https://github.com/user-attachments/assets/6e297869-3d5f-4584-9a2d-39df13cc6acb">

Naming the file `login.tsx` would return an unmatched route error because it is expected that an index. TSX file is within this layout unless there is an `index.tsx` file in the root. To ensure that the unauthenticated user does not access the app, we want to make sure that only the pages within the (auth) are reachable before authentication.


If we want to keep the structure to have the `(auth)/login.tsx` file. What we may do is use the `app/index.tsx` as the onboarding page which will be the landing page for users when opened the app

Example of an Oboarding page
<img width="303" alt="Screenshot 2024-11-03 at 12 24 23" src="https://github.com/user-attachments/assets/41f5b454-a756-472d-a5cd-418923149edb">

<img width="1920" alt="Screenshot 2024-11-03 at 12 18 08" src="https://github.com/user-attachments/assets/1f0f003f-5a76-4278-bcc4-bf916918d6af">

Reply:
Before jumping to a solution, let's analyze the problem first

What did you try to change the route? Did you also update the stack screen config?

Reply:
Yes i did `<Stack.Screen name="(auth)/login" options={{ headerShown: false }} />` 

The App needs a page to load in first render